packages/kbn-build/src/utils/npm.js

```diff
@@ -0,0 +1,46 @@
+import { spawn, spawnStreaming } from './childProcess';
+
+export function runScriptInDir(script, args, directory) {
+  return spawn('npm', ['run', script, ...args], {
+    cwd: directory
+  });
+}
+
+export function runScriptInPackageStreaming(script, args, pkg) {
+  const execOpts = {
+    cwd: pkg.path
+  };
+
+  const stream = spawnStreaming('npm', ['run', script, ...args], execOpts, {
+    prefix: pkg.name
+  });
+
+  // TODO Add a timeout that triggers in case we're not able to see a
+  // completion trigger.
+  const started = new Promise(resolve => {
+    stream.stdout.on('data', resolveOnStartup);
+
+    function resolveOnStartup(data) {
+      const isTypeScriptReady = data.includes('Compilation complete.');
```

Comment:
__note:__ these checks look fragile, but I don't have a better suggestion at the moment and we can improve it later :)